bitnami/keycloak/templates/configmap-env-vars.yaml

```diff
@@ -24,6 +24,12 @@ data:
   KEYCLOAK_DATABASE_NAME: {{ include "keycloak.databaseName" . | quote }}
   KEYCLOAK_DATABASE_USER: {{ include "keycloak.databaseUser" . | quote }}
   {{- end }}
+  {{- if (not .Values.postgresql.enabled) }}
+  KEYCLOAK_DATABASE_HOST: {{ .Values.externalDatabase.host | quote }}
+  KEYCLOAK_DATABASE_PORT: {{ .Values.externalDatabase.port | quote }}
+  KEYCLOAK_DATABASE_NAME: {{ .Values.externalDatabase.database | quote }}
+  KEYCLOAK_DATABASE_USER: {{ .Values.externalDatabase.user | quote }}
+  {{- end }}
```

Comment:
The named templates defined in the `_helpers.tpl` already returns you the right host/port/database/user depending on whether you're using an external database or not. In case you want to follow with this approach, I'd simply remove the condition in line 21.